build_bundle.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/python3
+import json
+import configparser
+import sys
+import os
+import tempfile
+import glob
+import subprocess
+import shutil
+import time
+
+CONFIG_FILE_NAME = "xt-bundle-aos.cfg"
+
+BUNDLE_CONFIG_SECTION = "bundle_config"
+TOP_TEMPLATE = "top_template"
+COMPONENTS = "components"
+OPT_DEPLOY_DIR = "bundle_deploy_dir"
+OPT_TYPE = "type"
+OPT_ITEM_TEMPLATE = "item_template"
+OPT_DEPENDENCIES = "dependencies"
+
+SECT_PATH = "path"
+OPT_BASE = "workspace_base_dir"
+
+RUNTIME_DEPS = "runtimeDependencies"
+
+
+class BundleProcessor:
+    def __init__(self, cfg):
+        print("BundleProcessor")
+        self.__cfg = cfg
+
+        #create bundle path
+        self.__temp = tempfile.mkdtemp()
+        print("Created temp dir for bundle, {}".format(self.__temp))
+
+        top_template_path = cfg.get(BUNDLE_CONFIG_SECTION, TOP_TEMPLATE)
+        with open(top_template_path) as top_template:
+            self.__top_template = json.load(top_template)
+            print("got top_template {}".format(str(self.__top_template)))
+            for key, value in self.__top_template.items():
+                if cfg.has_option(BUNDLE_CONFIG_SECTION, key):
+                    self.__top_template[key] = cfg.get(
+                        BUNDLE_CONFIG_SECTION, key)
+
+            print("processed top_template {}".format(str(self.__top_template)))
+
+        self.__components = list()
+        self.__components_data = list()
+        for component in cfg.get(BUNDLE_CONFIG_SECTION,
+                                        COMPONENTS).split():
+            print("component = {}".format(component))
+            self.__components.append(component)
+            self.__load_component(component)
+        print("components = {}".format(self.__components))
+
+        #Processing runtimeDependencies
+        for comp in self.__components_data:
+            comp[RUNTIME_DEPS] = self.__process_deps(comp['id'])
+
+    def cleanup(self):
+        shutil.rmtree(self.__temp, ignore_errors=True)
+
+    def __process_deps(self, component):
+        if self.__cfg.has_option(component, OPT_DEPENDENCIES) is not True:
+            return {}
+
+        deps = self.__cfg.get(component, OPT_DEPENDENCIES)
+        deps_template = list()
+        for dep in deps.split():
+            print("got dep {}".format(dep))
+            dep_list = dep.split("|")
+            if dep_list[0] in self.__components:
```

Comment:
Provide some error checking (for example, `len(dep_list) == 2`) and raise human readable error (instead of `index of range` if dep list has inappropriate format)